src/servers/database.ts

```diff
@@ -550,7 +588,49 @@ export class DatabaseServer extends SkeletonServer {
       return null;
     }
   }
+  async eth_getTransactionReceiptsByBlockNumber(
+    _request: any,
+    blockNumber: web3.Quantity | web3.Tag
+  ): Promise<web3.TransactionReceipt[] | null> {
+    const blockNumber_ =
+      parseBlockSpec(blockNumber) || (await this._fetchCurrentBlockID());
+    try {
+      const {
+        rows: [receipt],
+      } = await this._query(
+        `SELECT
+        b.id AS "blockNumber",
+        b.hash AS "blockHash",
+        t.index AS "transactionIndex",
+        t.hash AS "transactionHash",
+        t.from AS "from",
+        t.to AS "to",
+        t.gas_used AS "gasUsed",
+        0::u256 AS "cumulativeGasUsed", -- TODO: tally?
+        CASE WHEN t.to IS NULL OR t.to = '\\x0000000000000000000000000000000000000000' THEN t.output
+             ELSE NULL
+        END AS "contractAddress",
+        NULL AS "logs",                 -- TODO: fetch event.id[]
+        repeat('\\000', 256)::bytea AS "logsBloom",
+        CASE WHEN t.status THEN 1 ELSE 0 END AS "status",
+        t.near_hash AS "nearTransactionHash",
+        t.near_receipt_hash AS "nearReceiptHash"
+      FROM transaction t
+        LEFT JOIN block b ON t.block = b.id
+      WHERE b.id = $1`,
+        [blockNumber_]
+      );
+      //assert(receipt, 'receipt is not null');
 
+      receipt.logs = await this._fetchEvents(hexToBytes(receipt.transactionHash.toString()));
+      return exportJSON(receipt);
```

Comment:
This line returns only one receipt and the function should return a list of receipts.